### PR TITLE
feat(SWR): add swr image trigger resource

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
   add swr image trigger resource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
   add swr image trigger resource
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/swr/' TESTARGS='-run TestAccSwrImageTrigger_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/swr/ -v -run TestAccSwrImageTrigger_basic -timeout 360m -parallel 4
=== RUN   TestAccSwrImageTrigger_basic
=== PAUSE TestAccSwrImageTrigger_basic
=== CONT  TestAccSwrImageTrigger_basic
--- PASS: TestAccSwrImageTrigger_basic (19.15s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/swr       19.201s
```

### CODE REVIEW

**Comment by houpeng80** [comment]
make testacc TEST='./huaweicloud/services/acceptance/swr/' TESTARGS='-run TestAccSwrImageTrigger_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/swr/ -v -run TestAccSwrImageTrigger_basic -timeout 360m -parallel 4
=== RUN   TestAccSwrImageTrigger_basic
=== PAUSE TestAccSwrImageTrigger_basic
=== CONT  TestAccSwrImageTrigger_basic
--- PASS: TestAccSwrImageTrigger_basic (19.38s) 
PASS 
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/swr       19.430s

**Comment by houpeng80** [comment]
make testacc TEST='./huaweicloud/services/acceptance/swr/' TESTARGS='-run TestAccSwrImageTrigger_basic'                 
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/swr/ -v -run TestAccSwrImageTrigger_basic -timeout 360m -parallel 4 
=== RUN   TestAccSwrImageTrigger_basic 
=== PAUSE TestAccSwrImageTrigger_basic
=== CONT  TestAccSwrImageTrigger_basic
--- PASS: TestAccSwrImageTrigger_basic (19.28s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/swr       19.328s

File: docs/resources/swr_image_trigger.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+  Changing this parameter will create a new resource.
+
+* `enabled` - (Required, String) Specifies whether to enable the trigger. Value options **true**, **false**.
+
+* `name` - (Required, String, ForceNew) Specifies the trigger name.
+
+  Changing this parameter will create a new resource.
+
+* `condition_type` - (Required, String, ForceNew) Specifies the trigger condition type.
```
Comment (ShiChangkuo) [inline docs/resources/swr_image_trigger.md]: it's better to put `condition_type` and `condition_value` together.

File: docs/resources/swr_image_trigger.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+* `condition_value` - (Required, String, ForceNew) Specifies the trigger condition value. Value options:
+  + When condition_type is set to `all`, set this parameter to `.*`.
+  + When condition_type is set to `tag`, set this parameter to specific image tags separated by semicolons (;).
+  + When condition_type is set to `regular`, set this parameter to a regular expression.
+
+  Changing this parameter will create a new resource.
+
+* `enabled` - (Required, String) Specifies whether to enable the trigger. Value options **true**, **false**.
+
+* `name` - (Required, String, ForceNew) Specifies the trigger name.
```
Comment (ShiChangkuo) [inline docs/resources/swr_image_trigger.md]: it's better to put `name` before other parameters.

File: docs/resources/swr_image_trigger.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  Changing this parameter will create a new resource.
+
+* `condition_value` - (Required, String, ForceNew) Specifies the trigger condition value. Value options:
+  + When condition_type is set to `all`, set this parameter to `.*`.
+  + When condition_type is set to `tag`, set this parameter to specific image tags separated by semicolons (;).
+  + When condition_type is set to `regular`, set this parameter to a regular expression.
+
+  Changing this parameter will create a new resource.
+
+* `enabled` - (Required, String) Specifies whether to enable the trigger. Value options **true**, **false**.
```
Comment (ShiChangkuo) [inline docs/resources/swr_image_trigger.md]: can we change this parameter to optional, and set a default value?

File: docs/resources/swr_image_trigger.md (hunk 1 above)
Comment (houpeng80) [reply docs/resources/swr_image_trigger.md]: done

File: docs/resources/swr_image_trigger.md (hunk 2 above)
Comment (houpeng80) [reply docs/resources/swr_image_trigger.md]: done

File: docs/resources/swr_image_trigger.md (hunk 3 above)
Comment (houpeng80) [reply docs/resources/swr_image_trigger.md]: the default value of console is true, so set the default value to true also

**Pull request merged**